streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -608,13 +612,26 @@ private void prepareTopic(Map<String, InternalTopicMetadata> topicPartitions) {
                     throw new TopologyBuilderException(String.format("stream-thread [%s] Topic [%s] number of partitions not defined", streamThread.getName(), topic.name()));
                 }
 
-                internalTopicManager.makeReady(topic, numPartitions);
+                topicsToMakeReady.put(topic, numPartitions);
+                topicNamesToMakeReady.add(topic.name());
+            }
+
+            if (!topicsToMakeReady.isEmpty()) {
+                internalTopicManager.makeReady(topicsToMakeReady);
 
-                // wait until the topic metadata has been propagated to all brokers
-                List<PartitionInfo> partitions;
+                // wait until the topic metadata has been propagated to at least one broker
+                boolean allTopicsCreated = true;
                 do {
-                    partitions = streamThread.restoreConsumer.partitionsFor(topic.name());
-                } while (partitions == null || partitions.size() != numPartitions);
+                    Map<String, Integer> partitions = internalTopicManager.getNumPartitions(topicNamesToMakeReady);
```

Comment:
My preference here would be to extract this logic into a method like:
```
private boolean allTopicsCreated(final Set<String> topicNamesToMakeReady, final Map<InternalTopicConfig, Integer> topicsToMakeReady) {
        final Map<String, Integer> partitions = internalTopicManager.getNumPartitions(topicNamesToMakeReady);
        for (Map.Entry<InternalTopicConfig, Integer> entry : topicsToMakeReady.entrySet()) {
            final Integer numPartitions = partitions.get(entry.getKey().name());
            if (numPartitions == null || !numPartitions.equals(entry.getValue())) {
                return false;
            }
        }
        return true;
    }
```

and then have:
```
while(!allTopicsCreated(topicNamesToMakeReady, topicsToMakeReady) {
   // should we add a small sleep here?
}
```
I think it makes the code cleaner. Removes the temporary variable and the break (neither of which i like!)